src/ansys/pre_commit_hooks/tech_review.py

```diff
@@ -0,0 +1,253 @@
+#
+# The above copyright notice and this permission notice shall be included in all
+# copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+# SOFTWARE.
+"""Module for checking if a repository is compliant with required files in the technical review."""
+import argparse
+from datetime import date as dt
+import os
+import pathlib
+import re
+
+# import shutil
+from tempfile import NamedTemporaryFile
+
+import git
+from jinja2 import Environment, FileSystemLoader
+
+# from reuse._licenses import _load_license_list
+import toml
+
+from ansys.pre_commit_hooks.add_license_headers import check_same_content
+
+# Get current git repository
+git_repo = git.Repo(os.getcwd(), search_parent_directories=True)
+REPO_PATH = git_repo.git.rev_parse("--show-toplevel")
+"""Default repository path."""
+DOC_REPO_NAME = os.path.basename(REPO_PATH)
+"""Name of the repository."""
+HOOK_PATH = pathlib.Path(__file__).parent.resolve()
+"""Location of the pre-commit hook on your system."""
+LICENSES_JSON = os.path.join(HOOK_PATH, "..", "reuse", "resources", "licenses.json")
+"""JSON file containing licenses information."""
+DEFAULT_AUTHOR_MAINT_NAME = "ANSYS, Inc."
+"""Default name of project authors and maintainers."""
+DEFAULT_AUTHOR_MAINT_EMAIL = "pyansys.core@ansys.com"
+"""Default email of project authors and maintainers."""
+DEFAULT_START_YEAR = dt.today().year
+"""Default start year of the repository."""
+DEFAULT_LICENSE = "MIT"
+"""Default license of the repository"""
+
+
+def check_config_file(author_maint_name, author_maint_email):
+    """Check naming convention, version, author, and maintainer information."""
+    # Is zero when the configuration file is compliant and one when it is not compliant
+    non_compliant = 0
+
+    if os.path.exists(os.path.join(REPO_PATH, "pyproject.toml")):
+        non_compliant, project_name = check_pyproject_toml(
+            author_maint_name, author_maint_email, non_compliant
+        )
+    elif os.path.exists(os.path.join(REPO_PATH, "setup.py")):
+        non_compliant, project_name = check_setup_py(
+            author_maint_name, author_maint_email, non_compliant
+        )
+    else:
+        print("pyproject.toml and setup.py files do not exist")
+        print("Project name does not follow naming conventions")
+        print("Project version does not contain Semantic versioning")
+        print("Author and maintainer is not ANSYS, Inc.")
+        print("Author and maintainer email is not pyansys.core@ansys.com")
```

Comment:
I wouldn't inform about all this. You can't really tell - they are missing the main Python configuration file of a repo. That's what you can tell.